### PR TITLE
Repository housekeeping varia, python 3.8 tests

### LINKED ISSUE #424: Test against minimum versions of required packages
We have defined minimum versions for many of our required packages in our `requirements` directory.  However, to the best of my knowledge we do not explicitly test against the minimum versions to make sure that PlasmaPy works with them.  It would be really helpful to add a test to Travis CI or AppVeyer that specifically tests the minimum supported versions as listed in those files.

One thing that we should be careful of is that the version on PyPI for the `roman` package differs from the `roman.__version__` attribute.

It would be most helpful to set things up so that the minimum versions are taken directly from `automated-code-tests.txt` rather than us having to type in and update the minimum versions somewhere else manually.

Thanks!

### LINKED ISSUE #635: Add python 3.8 to testing matrix

### PR DESCRIPTION
**Pull request by StanczakDominik**
This implements https://github.com/OpenAstronomy/azure-pipelines-templates for CI.

Closes #635, closes #424.

### CODE REVIEW

**Comment by StanczakDominik** [comment]
The py37-astropydev error is a result of #653 and I'm still not sure how to handle it

I will also add #424 here

**Comment by StanczakDominik** [comment]
- [ ] `py36-minimal-conda [linux])` is failing due to calls to `units.quantity.allclose`, which either wasn't there yet or was in another namespace back in Astropy 2.0. But I think simply bumping the minimum to 3.0, or even 3.1 as I seem to remember some breakage, would be more sensible.
- [ ] `py37-astropydev [linux]` is all #653, and I think it could be possible to fix most of it via `atol`s. I'll try it tomorrow or so.

I'm also liking how local access to these envs is more comfortable now.

**Comment by StanczakDominik** [comment]
Also, @Cadair, consider this tested and I've got only positive feedback for you :) it seems to be working like a charm.

**Comment by StanczakDominik** [comment]
I've decided to bump the minimum required versions of

- Astropy from 2.0 to 3.1 because Seriously, That's One Major Version Ago and Let's Respect Each Other
- Numpy from 1.13 to 1.14 because https://github.com/PlasmaPy/PlasmaPy/issues/229 And That Should Be Enough Said

**Comment by namurphy** [comment]
As much as I'll miss broken doctests, bumping the minimum version of NumPy to 1.14 sounds like a good idea to me.  Same for bumping minimum Astropy to 3.1.  Thanks!

**Comment by StanczakDominik** [comment]
Turns out the astropydev problem consists of a bunch of different problems and I'd much rather merge this now, then figure out fixes for each...

**Comment by StanczakDominik** [comment]
ELLIPSIS seems broken within astropy/pytest-doctestplus, I'll try to report that there, but most definitely not tonight.

For now, I'm instead going to get rid of pytest-astropy, as I'm realizing we're not actually using any of its functionality except FLOAT_CMP, and that can be reproduced with `pytest >= 5.1` via the NUMBER flag.

I'm frustrated, so I shall now vent with a witty video:

https://www.youtube.com/watch?v=djZkTnJnLR0

**Comment by StanczakDominik** [comment]
https://www.youtube.com/watch?v=AXwGVXD7qEQ

except for 3.8 failing on pip installing hdf5, and astropy-dev is still failing, but that's still PROGRESS!

**Comment by Cadair** [comment]
I *think* you might be able to solve your hdf5 issue by installing the hdf5 c lib with apt.

**Comment by StanczakDominik** [comment]
I think I'll just take the lazy route and wait for https://github.com/h5py/h5py/issues/1410 :laughing:

**Comment by Cadair** [comment]
That looks like a troublesome bug in the Azure images...

File: docs/install.rst [hunk 1]
```diff
@@ -19,9 +19,9 @@ PlasmaPy requires Python version 3.6 or newer, and is not compatible
 with Python 2.7.  PlasmaPy requires the following packages for
 installation:
 
-- `NumPy <http://www.numpy.org/>`_ 1.13 or newer
+- `NumPy <http://www.numpy.org/>`_ 1.14 or newer
```
Comment (namurphy) [inline docs/install.rst]: And there was much rejoicing
